lib/rex/proto/dns/resolver.rb

```diff
@@ -0,0 +1,312 @@
+require 'net/dns/resolver'
+
+module Rex
+module Proto
+module DNS
+
+  ##
+  # Provides Rex::Sockets compatible version of Net::DNS::Resolver
+  ##
+  class Resolver < Net::DNS::Resolver
```

Comment:
This is all a tire fire, but due to Net::DNS, not this PR specifically. Do those /dev/null filenames cause any weirdness on Windows?